src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -56,14 +59,20 @@ public virtual IDisposable Start()
 
             var applicationDelegate = BuildApplicationDelegate();
 
-            var _contextFactory = _applicationServices.GetRequiredService<IHttpContextFactory>();
-            var _contextAccessor = _applicationServices.GetRequiredService<IHttpContextAccessor>();
+            var logger = _applicationServices.GetRequiredService<ILogger<HostingEngine>>();
+            var contextFactory = _applicationServices.GetRequiredService<IHttpContextFactory>();
+            var contextAccessor = _applicationServices.GetRequiredService<IHttpContextAccessor>();
             var server = _serverFactory.Start(_serverInstance,
                 features =>
                 {
-                    var httpContext = _contextFactory.CreateHttpContext(features);
-                    _contextAccessor.HttpContext = httpContext;
-                    return applicationDelegate(httpContext);
+                    var httpContext = contextFactory.CreateHttpContext(features);
+                    var requestIdentifier = GetRequestIdentifier(httpContext);
+
+                    using (logger.BeginScope("Request Id: {RequestId}", requestIdentifier))
+                    {
+                        contextAccessor.HttpContext = httpContext;
+                        return applicationDelegate(httpContext);
```

Comment:
No, I meant that `return applicationDelegate(httpContext);` exits the using statement as soon as any Task is returned from the applicationDelegate. You need to use await to make sure the returned task actually finishes before you close the scope.
The changes look good. :shipit: